src/Raven.Studio/typescript/viewmodels/resources/resources.ts

```diff
@@ -247,9 +254,15 @@ class resources extends viewModelBase {
         }
     }
 
-    private removeResource(resource: resourceInfo) {
-        this.resources().sortedResources.remove(resource);
-        this.selectedResources.remove(resource.qualifiedName);
+    private removeResource(rsInfo: resourceInfo) {
+        this.resources().sortedResources.remove(rsInfo);
+        this.selectedResources.remove(rsInfo.qualifiedName);
+
+        let resource = this.resourcesManager.resources().find(x => x.qualifiedName === rsInfo.qualifiedName);
+        if (!resource)
+            return;
+
+        ko.postbox.publish(EVENTS.Resource.Disconnect, { resource: resource });
```

Comment:
I'm not sure call publishing this event in resources is good idea. I will have a look at this as a part of: RavenDB-5967